nutils/evaluable.py

```diff
@@ -2313,12 +2336,15 @@ class Elemwise(Array):
   @types.apply_annotations
   def __init__(self, data:types.tuple[types.frozenarray], index:asarray, dtype:asdtype):
     self.data = data
+    # tuple(d.astype(dtype, copy=False) for d in data)
+    print('data', data)
+    print('fuck', tuple(d.dtype for d in data), dtype)
     shape = get([d.shape for d in data], 0, index)
     super().__init__(args=[index], shape=shape, dtype=dtype)
 
   def evalf(self, index):
     assert index.ndim == 0
-    return self.data[index]
+    return self.data[index].astype(self.dtype, copy=False)
```

Comment:
I'd add argument `casting=safe`to `astype`. I don't think it is desirable to allow casting complex to float or float to int etc.

Reply:
This triggers an error on windows where `np.int32` is the default dtype for win64.